cpacker/MemGPT
# Fix max tokens constant

Closes #352, #353

**Please describe the purpose of this pull request.**
Patch #352

This PR touches:

- agent configs (test save/load)
- local LLM calls (test basic completion)
  - make sure that local LLMs get context length set to 8k by default, and that eg gpt-4-turbo gets set to 128k
- openai LLM calls (implicitly) (test openai functionality)
- summarizer calls (test summarize)

**How to test**

- With OpenAI: try `memgpt configure`, and it should autoselect the context window length
  - Then check that it's set correctly in config
- With local models: try `memgpt configure`, and it should ask for the context window length
  - Then check that it's set correctly in config
- With local models: try `memgpt run --context-window <N>`, and it should override the context window length
  - Check in config
  - But also test via local inference

**Have you tested this PR?**
Yes

**Related issues or PRs**
#352, #353

**Is your PR over 500 lines of code?**
n/a

**Additional context**
n/a
